src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Emitter.cs

```diff
@@ -65,29 +65,60 @@ private static bool UseLoggerMessageDefine(LoggerMethod lm)
 
             private void GenType(LoggerClass lc)
             {
+                int indentationSize = 0;
+                string nestedIndentation = "";
                 if (!string.IsNullOrWhiteSpace(lc.Namespace))
                 {
                     _builder.Append($@"
 namespace {lc.Namespace}
 {{");
                 }
 
+                LoggerClass parent = lc.ParentClass;
+                Stack<string> parentClasses = new Stack<string>();
```

Comment:
Since you don't really "push" and "pop" things on and off this stack, you could just use a `List<string>` and then walk over it in the reverse order `for (int i = parentClasses.Length - 1; i >= 0; i--)`.